lib/std/zig/render.zig

```diff
@@ -1159,7 +1159,13 @@ fn renderContainerField(
         try renderToken(ais, tree, rparen_token, .space); // )
     }
     const eq_token = tree.firstToken(field.ast.value_expr) - 1;
-    try renderToken(ais, tree, eq_token, .space); // =
+    const eq_space: Space = if (tree.tokensOnSameLine(eq_token, eq_token + 1)) .space else .newline;
+    {
+        ais.pushIndent();
+        try renderToken(ais, tree, eq_token, eq_space); // =
+        ais.popIndent();
+    }
+    ais.pushIndentOneShot();
```

Comment:
A "oneshot" indent won't work properly in the following case:
```zig
const Foo = struct {
    bar: u32 =
        blk: {
            break :blk 42;
        },
};
```
I think the indentation we want here would be:
```zig
ais.pushIndentNextLine();
renderToken(...);
renderExpressionComma(...);
ais.popIndent();
```
Note this is broken in the same way for assignment currently  #7363. It looks pretty ugly to fix though :/

Reply:
Pushed a fix for that case. Yeah, it gets ugly pretty quick :\